### PR TITLE
C# extractor code cleanup

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

**Comment by tamasvajk** [comment]
C# differences job: https://jenkins.internal.semmle.com/job/Changes/job/CSharp-Differences/507/

**Comment by tamasvajk** [comment]
C# differences job failed again with 
```
The SSL connection could not be established, see inner exception.
   Authentication failed, see inner exception.
   Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED
     at /build/mono-6.12.0.90/external/boringssl/ssl/handshake_client.c:1132
```

**Comment by tamasvajk** [comment]
https://jenkins.internal.semmle.com/job/Changes/job/CSharp-Differences/516/

**Comment by tamasvajk** [comment]
@hvitved FYI, the C# differences job passes on this one.

File: csharp/autobuilder/Semmle.Autobuild.CSharp/CSharpAutobuilder.cs [hunk 1]
```diff
@@ -104,7 +107,9 @@ private CSharpBuildStrategy GetCSharpBuildStrategy()
                 || Options.MsBuildConfiguration != null
                 || Options.MsBuildPlatform != null
                 || Options.MsBuildTarget != null)
+            {
                 return CSharpBuildStrategy.MSBuild;
```
Comment (hvitved) [inline csharp/autobuilder/Semmle.Autobuild.CSharp/CSharpAutobuilder.cs:111]: Not sure why braces were added here, as the `return` is not a multi-line statement.

**Review by hvitved — COMMENTED** [review]
Overall this looks really great. Perhaps you could write a guide how to setup VS Code to respect the formatting settings?

I didn't look at everything in full detail, but I am quite confident that the changes are OK given successful tests and CSharp-Differences.

File: csharp/autobuilder/Semmle.Autobuild.CSharp/CSharpAutobuilder.cs (hunk 1 above)
Comment (tamasvajk) [reply csharp/autobuilder/Semmle.Autobuild.CSharp/CSharpAutobuilder.cs:111]: The condition is multiline, and that's also taken into account. See here: https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Analyzers/CSharp/Analyzers/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs#L169-L180

File: csharp/autobuilder/Semmle.Autobuild.Shared/AutobuildOptions.cs [hunk 2]
```diff
@@ -9,26 +9,24 @@ namespace Semmle.Autobuild.Shared
...
-        public readonly string? BuildCommand;
-        public readonly string[] Solution;
-
-        public readonly bool IgnoreErrors;
-        public readonly bool Buildless;
-        public readonly bool AllSolutions;
-        public readonly bool NugetRestore;
-
-        public readonly Language Language;
+        public int SearchDepth { get; } = 3;
```
[the remark this answers is not in the document]
Comment (tamasvajk) [reply csharp/autobuilder/Semmle.Autobuild.Shared/AutobuildOptions.cs:15]: A field is part of the internal storage representation of a class and the encapsulation principle says that internals of a class should be hidden from the outside, so basically all fields should be `private` (or maybe `protected`). I guess in some sense this is philosophical. But in general this is the suggested best practice too. Check paragraph 3 of this page: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/fields.

There are cases when properties have definite benefits. For example in UI based applications you can bind to properties, but not to fields, and can get notifications on updates. In our case, I think the only major benefit is that accessibility of `get` and `set` can be controlled separately for properties, which gives us finer control on who has read/write access to what. We used this for example in `Semmle.Extraction.CommonOptions`.

**Comment by tamasvajk** [comment]
> Overall this looks really great. Perhaps you could write a guide how to setup VS Code to respect the formatting settings?

@hvitved I think this should happen automatically when opening the `csharp` folder in VSCode. The settings in `.vscode/settings.json` turn on the required features in omnisharp. There's no way to force vscode to show all issues, and sometimes it takes quite long until they show up in the problems window.

I think eventually `dotnet format` global tool will be able to list all issues, but currently only a subset of editor config rules are supported.

**Review by hvitved — APPROVED** [review]

**Pull request merged**